src/store/participantsStore.js

```diff
@@ -288,20 +284,19 @@ const mutations = {
 	 *        participant.
 	 * @param {boolean} data.typing - whether the participant is typing or not.
 	 */
-	setTyping(state, { token, sessionId, typing }) {
-		if (!typing) {
-			if (state.typing[token] && state.typing[token].indexOf(sessionId) >= 0) {
-				state.typing[token].splice(state.typing[token].indexOf(sessionId), 1)
-			}
-
-			return
+	setTyping(state, { token, sessionId, typing, expirationTimeout }) {
+		if (!state.typing[token]) {
+			Vue.set(state.typing, token, {})
 		}
 
-		if (!state.typing[token]) {
-			Vue.set(state.typing, token, [])
+		if (state.typing[token][sessionId]) {
+			clearTimeout(state.typing[token][sessionId].expirationTimeout)
 		}
-		if (!state.typing[token].includes(sessionId)) {
-			state.typing[token].push(sessionId)
+
+		if (typing) {
+			Vue.set(state.typing[token], sessionId, { expirationTimeout })
+		} else {
+			Vue.delete(state.typing[token], sessionId)
 		}
```

Comment:
It seems to me, that the problem here was only with `if (!typing)` condition.

The "this.typing" state in signaling is inconsistent with store state. This state should be only in the store state.